gnucash/import-export/ofx/gnc-ofx-import.c

```diff
@@ -1038,6 +1025,95 @@ double ofx_get_investment_amount(const struct OfxTransactionData* data)
     }
 }
 
+// gnc_ofx_process_next_file processes the next file in the info->file_iter list.
+static void
+gnc_ofx_process_next_file (GtkDialog *dialog, gpointer user_data)
+{
+    ofx_info* info = (ofx_info*) user_data;
+    if (info->file_iter->next)
+    {
+        // Done with the previous OFX file, process the next one if any.
+        GSList* temp = info->file_iter;
+        info->file_iter = info->file_iter->next;
+        temp = g_slist_delete_link(temp, temp);
+        gnc_file_ofx_import_process_file (info);
```

Comment:
Likely you can shorten this whole list juggling to just this:

```
    ofx_info* info = (ofx_info*) user_data;
    info->file_iter = g_slist_delete_link(info->file_iter, info->file_iter);
    if (info->file_iter)
        gnc_file_ofx_import_process_file (info);
    else
        ...
```

g_slist_delete_link returns the new head of the GSList after the requested element has been deleted.

However I also think both options still have a string memory leak. The filenames stored in the GSList are not freed by g_slist_delete_link. It will only free the GSList data structure. So you probably want to add a g_free before calling g_slist_delete_link.

Lastly your function assumes it will never be called with an info->file_iter that's set to NULL. While the code you have now written probably adheres to that constraint it's fragile to just assume this to be the case forever. So I propose to add a guard upfront checking that both info and info->file_iter are not NULL before any dereferencing.

Reply:
Yes, I don't know how consistently that's done in GC, sometime I see it done, sometimes not. But I can certainly add that! I'll check the freeing stuff, thanks for the review.

Reply:
It's not done very consistently everywhere indeed. I think in this case it would be relatively safe to assume you are never getting a NULL info parameter. I'm more worried about dereferencing info->file_iter in the future as it's less obvious.

Reply:
Oh, one more nitpick; you can call it `file_list` rather than `file_iter`. You're not really using it to iterate in a loop.